Source/objects.cpp

```diff
@@ -1288,27 +1288,33 @@ void AddBarrel(Object &barrel)
 void AddShrine(Object &shrine)
 {
 	shrine._oRndSeed = AdvanceRndSeed();
-	bool slist[NumberOfShrineTypes];
-
 	shrine._oPreFlag = true;
 
-	int shrines = gbIsHellfire ? NumberOfShrineTypes : 26;
+	int shrineCount = gbIsHellfire ? NumberOfShrineTypes : 26;
+	bool slist[NumberOfShrineTypes] = { false };

```

Comment:
you can't do this

Reply:
I'm not sure I understand what you mean?
`bool slist[NumberOfShrineTypes];` = correct
bool slist[NumberOfShrineTypes] = { false };` = incorrect ?
I'm just initializing it to false.

Reply:
I think that by rearranging the code, you've tricked him into thinking you used `shrineCount` as the array size.

Reply:
> I think that by rearranging the code, you've tricked him into thinking you used `shrineCount` as the array size.

yup haha